modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -1275,11 +1282,16 @@ private CompletableFuture<Void> createTableLocally(
 
             startedTables.put(tableId, table);
 
-            return allOf(localPartsUpdateFuture, tablesByIdFuture)
-                    .thenComposeAsync(ignore -> inBusyLock(
-                            busyLock,
-                            () -> startLocalPartitionsAndClients(assignmentsFuture, table, zoneDescriptor.id())
-                    ), ioExecutor);
+            return allOf(localPartsUpdateFuture, tablesByIdFuture).thenComposeAsync(ignore -> inBusyLock(busyLock, () -> {
+                        if (onNodeRecovery) {
```

Comment:
Are you sure that we only do this on recovery?
How are we creating PK index on new table then? This code looks suspicious.